icu4j/main/classes/core/src/com/ibm/icu/impl/number/LongNameHandler.java

```diff
@@ -346,12 +346,19 @@ static void getMeasureData(
         subKey.append(unit.getType());
         subKey.append("/");
 
+        //TODO(ICU-21938): remove after fixing the data in CLDR-15421
```

Comment:
CLDR-15241 is not going to be fixed, CLDR is behaving as intended, see comments in that ticket. Any fix here needs to be a real fix, not a hack.

Reply:
Yes, the problem is that ICU is not using the alias mapping from CLDR (see https://github.com/unicode-org/cldr/blob/main/common/supplemental/units.xml#L485) in order to map deprecated unit identifiers to the new ones.

Reply:
Okay, no we need to map the aliases before we look to it. However, those data is not in added yet and I need to added. see

Reply:
I will try to fix that mapping of alias data when I do the CLDR  beta1 to ICU integration starting today.

Reply:
I already wrote the rules to add the aliases, I am still figuring out how to run the tool.
https://github.com/unicode-org/icu/pull/2026

I will try to finish it by tomorrow and you can review it if you do not mind.